packages/wallet/src/services/WalletUtil.ts

```diff
@@ -31,6 +31,61 @@ export const createInputResolver = ({ utxo }: InputResolverContext): Cardano.Inp
   }
 });
 
+/**
+ * Creates a decorator for a Cardano input resolver that enhances its functionality
+ * by adding the capability to fetch transaction inputs from the backend if they are
+ * not resolved by the original input resolver.
+ *
+ * This decorator function utilizes a provided `inputResolver` to attempt to resolve
+ * transaction inputs. If the `inputResolver` cannot find the input, the function then
+ * tries to fetch the transaction from the backend using a `ChainHistoryProvider`. It
+ * also caches fetched transactions to optimize subsequent input resolutions.
+ *
+ * @param inputResolver The original input resolver to be decorated.
+ * @param provider The backend provider used to fetch transactions by their hashes if
+ * they are not found by the inputResolver.
+ * @param localTxs An optional array of transactions to search before querying the backend.
+ * @returns A decorated input resolver with enhanced functionality to fetch unresolved inputs from the backend.
+ */
+export const createBackendInputResolver = (
+  inputResolver: Cardano.InputResolver,
+  provider: ChainHistoryProvider,
+  localTxs: Cardano.Tx[] = []
```

Comment:
How about this instead? 
```ts
const resolver = combineInputResolvers(walletUtil, createBackendInputResolver)
```

This way we could have a resolver that lives longer (not recreated every time) and has a more effective cache. We already have a local InputResolver in WalletUtil so we could just pass that here.

> I am thinking this could be useful for transaction chaining in the future (or to have the resolver look into any transaction not yet on chain)

That makes sense, we could then add an optional parameter to `resolveInput` function. There's been some discussion about adding this parameter to cip30 signTx.

Reply:
done here: https://github.com/input-output-hk/cardano-js-sdk/pull/1082/commits/d0dc28827485d2744c7fd9aba045a3066c2e4417 and here: https://github.com/input-output-hk/cardano-js-sdk/pull/1082/commits/c90750bed3c30425a8237f053577d980229dc926